### PR TITLE
Improved error handling while traversing

### PR DESCRIPTION
**Pull request by RayProud**
Fixed some bugs and improve error handling in traversing:
- Changed error messages — now it's more clear what was wrong with configurations,
- Add options check in List and ListToggle components,
- Handled a case when there were both `section.then` and `response.then`, but such key as `response.then` wasn't found in `section.then` (I found it a very useful case during my own project where I used Wizard),
- Changed `validateSection` because it didn't check all must-have properties,
- Changed try-catch to .catch, because try-catch couldn't handle promise errors.

### CODE REVIEW

File: src/components/Component.js [hunk 1]
```diff
@@ -76,6 +76,7 @@ class Component {
 
   /** Exit gracefully. */
   cleanAndExit() {
+    this.newline();
```
Comment (RayProud) [inline src/components/Component.js]: Without a newline errors disappear.

Comment (AaronGillBraun) [reply src/components/Component.js]: Oh apologies. I was going to look at your PR this morning but I got busy and forgot. I'll do by best to review it tomorrow. I took a quick glance and it looks good though.

Comment (RayProud) [reply src/components/Component.js]: Not a problem!
Take your time.

File: src/components/Wizard.js [hunk 2]
```diff
@@ -87,33 +97,53 @@ class Wizard extends Component {
...
           }
-          this.write(this.ansi.cursorShow);
-          reject(new Error('An unknown error occured'));
-          return null;
+
+          if (!section.then[response.then]) {
+            this.write(this.ansi.cursorShow);
+            reject(
+              new Error(
+                `Invalid configuration — couldn't find key '${
```
Comment (AaronGillBraun) [inline src/components/Wizard.js]: Same as above

File: src/components/Wizard.js [hunk 3]
```diff
@@ -87,33 +97,53 @@ class Wizard extends Component {
...
+                  response.then
+                }' in 'then' of '${section.id}' section`,
+              ),
+            );
+            return null;
+          }
+        } catch (err) {
+          reject(`A component error occured — ${err}`);
```
Comment (AaronGillBraun) [inline src/components/Wizard.js]: This isn't as much as a suggestion as it is a question: Should we start to sort of "group" errors into their specific error types? Similar to how configuration errors are prefixed with `Invalid configuration - `, should we have `Component error - `? Or should we use a different prefix? This is more of a fatal error than anything but still... Anyway, just something to think about.

File: src/components/Wizard.js [hunk 4]
```diff
@@ -87,33 +97,53 @@ class Wizard extends Component {
...
         }
+
         this.write(this.ansi.cursorShow);
-        reject(new Error('Invalid configuration'));
+        reject(new Error('An unknown error occured'));
         return null;
-      }).then(results => {
+      }
+      this.write(this.ansi.cursorShow);
+      reject(new Error('Invalid configuration'));
```
Comment (AaronGillBraun) [inline src/components/Wizard.js]: `Invalid configuration - Unknown component type '${section.type}' referenced`.

File: src/components/List.js [hunk 5]
```diff
@@ -6,13 +6,17 @@ class List extends Component {
...
     this.question = question.question;
+    this.questionId = question.id;
     this.options = question.options;
 
     this.chosen = 0;
   }
 
   init() {
     return new Promise(resolve => {
+      if (!this.options || !this.options.length)
```
[the remark this answers is not in the document]
Comment (RayProud) [reply src/components/List.js]: I totally agree with you. But it was prettier that made it.
I'll fix it.

File: src/components/List.js [hunk 6]
```diff
@@ -6,13 +6,17 @@ class List extends Component {
...
+        this.throw(`couldn't find options in question '${this.questionId}'`);
```
[the remark this answers is not in the document]
Comment (RayProud) [reply src/components/List.js]: Got it.

File: src/components/List.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (AaronGillBraun) [reply src/components/List.js]: Oh really? I've never had prettier do that to me. Maybe i'll check to see which setting caused that.

File: src/components/Wizard.js (hunk 3 above)
Comment (RayProud) [reply src/components/Wizard.js]: I'm not really sure about now.
Options problem is an invalid configuration case, so maybe just not now.

In this case I don't mind use just `Invalid configuration`.

File: src/components/Wizard.js [hunk 7]
```diff
@@ -63,21 +63,31 @@ class Wizard extends Component {
...
-          throw new Error('Invalid configuration');
-        }
-        const ComponentType = this.components[section.type];
-        if (ComponentType) {
+  traverse(section) {
+    return new Promise(async (resolve, reject) => {
+      if (!this.validateSection(section)) {
+        reject(
+          new Error(
+            `Invalid configuration — check the docs to ensure the question was written by the rules`,
```
[the remark this answers is not in the document]
Comment (RayProud) [reply src/components/Wizard.js]: Ok!

File: src/components/ListToggle.js [hunk 8]
```diff
@@ -6,14 +6,18 @@ class ListToggle extends Component {
...
     this.options = question.options;
 
-    this.selected = new Array(this.options.length);
+    this.selected = new Array(this.options && this.options.length);
     this.chosen = 0;
   }
 
   init() {
     return new Promise(resolve => {
+      if (!this.options || !this.options.length)
```
[the remark this answers is not in the document]
Comment (RayProud) [reply src/components/ListToggle.js]: Fixed.

File: src/components/List.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (RayProud) [reply src/components/List.js]: Fixed.

File: src/components/Wizard.js [hunk 9]
```diff
@@ -87,33 +97,53 @@ class Wizard extends Component {
           }
           if (!section.then) {
             this.write(this.ansi.cursorShow);
-            reject(new Error('Invalid configuration'));
+            reject(
+              new Error(
+                `Invalid configuration — couldn't find 'then' key in '${
```
[the remark this answers is not in the document]
Comment (RayProud) [reply src/components/Wizard.js]: Cool! Thanks for helping with my grammar. 
Fixed!

File: src/components/Wizard.js (hunk 2 above)
Comment (RayProud) [reply src/components/Wizard.js]: Fixed.

File: src/components/Wizard.js (hunk 4 above)
Comment (RayProud) [reply src/components/Wizard.js]: Fixed.

**Comment by RayProud** [comment]
Thanks for reviewing!
Will you merge it? I'd like to have these commits to implement `text` type.

**Comment by AaronGillBraun** [comment]
Whoops, thought I merged it. Sorry :)

**Comment by RayProud** [comment]
😀

**Pull request merged**